src/context/WalletProvider/NativeWallet/components/NativeSuccess.tsx

```diff
@@ -1,12 +1,16 @@
 import { Box, ModalBody, ModalHeader } from '@chakra-ui/react'
+import { useQueryClient } from '@tanstack/react-query'
 import { Text } from 'components/Text'
 
 import { useNativeSuccess } from '../hooks/useNativeSuccess'
 import type { NativeSetupProps } from '../types'
 
 export const NativeSuccess = ({ location }: NativeSetupProps) => {
+  const queryClient = useQueryClient()
   const { isSuccessful } = useNativeSuccess({ vault: location.state.vault })
 
+  queryClient.invalidateQueries({ queryKey: ['hdwalletNativeVaultsList'] })
```

Comment:
Invalidates query on success since this is now a query vs. ugly IIAFE (with the flag on) - noop with the flag off

Reply:
waddabout 3234de0c84060f155fd82794057399e3cd854705